Adding static schema for policy resource

RDavisHPE opened the pull request:
Adding static schema for policy resource so that populating the config is easier and more intuitive than figuring out how to use the dynamic config.

RDavisHPE committed:
Adding static schema for policy resource

Adding static schema for policy resource so that populating the config is
easier and more intuitive than figuring out how to use the dynamic config.

RDavisHPE committed:
Added numbering comments to mapStateToAddPolicyConfig

RDavisHPE committed:
Fixed motd

RDavisHPE committed:
Fixed lifecycle (missing account integration id)

RDavisHPE committed:
Fixed shutdown

RDavisHPE committed:
Fixed on/off conversions

RDavisHPE committed:
Fixed linting

RDavisHPE committed:
Ran make docs

RDavisHPE committed:
Testing all static schema

RDavisHPE committed:
Fixed lint error

RDavisHPE commented on internal/framework/subproviders/morpheus/convert/convert.go:
```diff
@@ -136,6 +137,23 @@ func Int64ToType(i *int64) types.Int64 {
 	return types.Int64Value(*i)
 }
 
+// StrToNumber converts a string pointer to a types.Number.
```
Can remove this once I update the OpenAPI spec

RDavisHPE replied:
https://github.com/HPE/terraform-provider-hpe/pull/646 Depends on this bump to get max_price as a number

RDavisHPE replied:
Bumped :+1:

RDavisHPE commented on internal/framework/subproviders/morpheus/resources/policy/create_test.go:
```diff
@@ -805,3 +805,854 @@ resource "hpe_morpheus_network" "test" {
 	})
 }
+
+// Test creating policies using static schema fields (config_* attributes)
```
Currently testing all 31 config options. Takes around 5 minutes to run. Will strip this down into something more manageable.

RDavisHPE replied:
Stripped down to two - since another test already works with all bare metal resources.

RDavisHPE committed:
Added conflicts_with between configs and flow ids

RDavisHPE committed:
Added ValidateConfig to ensure correct flow id is set.

RDavisHPE committed:
Removed comma and fixed linting

RDavisHPE committed:
Update schema with descriptions

RDavisHPE committed:
Using override descriptions

RDavisHPE committed:
Removed unnecessary null initialisation

RDavisHPE committed:
Commented out excess tests

RDavisHPE committed:
Actually lets just remove those tests

RDavisHPE committed:
Adding conflicts_with config and tests

RDavisHPE committed:
Switch to NumToType

RDavisHPE committed:
Merge branch 'dev-0.4' into policy-rs-schemas

RDavisHPE committed:
Added new number and set descriptions

RDavisHPE committed:
Merge branch 'dev-0.4' into policy-rs-schemas

matthew-c-hpe commented on internal/framework/subproviders/morpheus/resources/policy/config.go:
```diff
@@ -0,0 +1,658 @@
+		}
+
+		// 2. config_backup_storage -> BackupTargetsPolicyTypeConfiguration
+		if !plan.ConfigBackupStorage.IsNull() {
```
Include an `!IsUnknown` check too like you did with 1. config_approval for this and all below

jasonquigleyhpecom replied:
Definitely, I've been bitten by this

RDavisHPE replied:
Added :+1:

RDavisHPE committed:
Removed tests

RDavisHPE committed:
Fixing flow_id/workflow_id conflict

RDavisHPE committed:
Added IsUnknown checks

RDavisHPE committed:
Merge branch 'dev-0.5' into policy-rs-schemas

RDavisHPE committed:
Add switch statement for policy config population (#717)

Adding a switch statement based off the policy type code to populate the
policy config_* configs. This allows for state to only be concerned with
the relevant config fields for the given policy type.
- Switch config.go to use a switch statement for
mapStateToAddPolicyConfig (also applies for Update thanks to JSON trick)
- Switch read.go to use a switch statement for populating the config_*
fields based off policy type code.
- Updated schema_gen.go to use optional for config_* fields.

RDavisHPE committed:
Adding required fields to policy configs (#719)

Adding required fields to policy configs for an added layer of
validation in the schema.
- Updated required config attributes as required
- Confirmed state issues have been resolved through acceptance tests
- Ran make docs

RDavisHPE committed:
Added missing switch statement cases

RDavisHPE committed:
Merge branch 'dev-0.5' into policy-rs-schemas
